test/framework/shootmigrationtest.go

```diff
@@ -236,11 +236,17 @@ func (t *ShootMigrationTest) GetMachineDetails(ctx context.Context, seedClient k
 // GetPersistedSecrets uses the seedClient to fetch the data of all Secrets that have the `persist` label key set to true
 // from the Shoot's control plane namespace
 func (t *ShootMigrationTest) GetPersistedSecrets(ctx context.Context, seedClient kubernetes.Interface) (map[string]corev1.Secret, error) {
+	return GetPersistedSecrets(ctx, seedClient, t.SeedShootNamespace)
+}
+
+// GetPersistedSecrets uses the seedClient to fetch the data of all Secrets that have the `persist` label key set to true
+// from the Shoot's control plane namespace
+func GetPersistedSecrets(ctx context.Context, seedClient kubernetes.Interface, namespace string) (map[string]corev1.Secret, error) {
```

Comment:
A `client.Reader` would be enough for this function:
```suggestion
func GetPersistedSecrets(ctx context.Context, seedClient client.Reader, namespace string) (map[string]corev1.Secret, error) {
```